torch/csrc/utils/python_stub.h

```diff
@@ -0,0 +1,4 @@
+#pragma once
+
+struct _object;
+typedef _object PyObject;
```

Comment:
What is this?!

Reply:
```
    Include Python.h less using a new stub header.
    
    In many "non-Python" headers, we include Python.h because we need
    to declare a pointer to PyObject, and solely because of that.  It
    would be a lot better if we had a simpler version of Python.h that
    just declared PyObject available for pointers, without anything
    else.  This is what torch/csrc/utils/python_stub.h does.
    
    The good thing about not including Python.h is that it is easy to
    be warning-less; no more ugly insertions of Python.h on headers
    where it has no good reason to be.
```